fdcache/fdcache.go

```diff
@@ -155,3 +159,17 @@ func GetAndForgetFile(conn net.Conn) *os.File {
 	delete(cache, key)
 	return entry.Fp // Pass ownership to caller
 }
+
+// GetUUID returns the UUID for a passed-in connection.
+func GetUUID(conn net.Conn) (string, error) {
+	key := makekey(conn)
+	mutex.Lock()
+	defer mutex.Unlock()
+	entry, found := cache[key]
+	if !found {
+		return "", fmt.Errorf("fd not found")
```

Comment:
Could you make an `error` here using `errors.New`? (What is the difference between making an error using `errors.New` and using `fmt.Errorf`?)